File: packages/api/src/videos/videos.controller.ts
import {
  Body,
  Controller,
  Get,
  Param,
  ParseIntPipe,
  Patch,
  Query,
} from '@nestjs/common';
import { VideosService } from './videos.service';
import { QueryVideosDto } from './dto/query-videos.dto';
import { UpdateVideoTagsDto } from './dto/update-video-tags.dto';

@Controller('videos')
export class VideosController {
  constructor(private readonly videos: VideosService) {}

  @Get()
  list(@Query() q: QueryVideosDto) {
    return this.videos.findAll(q);
  }

  @Get('folders')
  folders(@Query('parent') parent = '') {
    return this.videos.getFolderListing(parent ?? '');
  }

  @Get(':id')
  one(@Param('id', ParseIntPipe) id: number) {
    return this.videos.findOne(id);
  }

  @Patch(':id/tags')
  patchTags(
    @Param('id', ParseIntPipe) id: number,
    @Body() body: UpdateVideoTagsDto,
  ) {
    return this.videos.updateTags(id, body.tagIds ?? []);
  }
}

File: packages/java/src/main/java/com/videomanager/videos/VideosController.java
package com.videomanager.videos;

import com.videomanager.videos.dto.QueryVideosDto;
import com.videomanager.videos.dto.UpdateVideoTagsDto;
import jakarta.validation.Valid;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PatchMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class VideosController {

    private final VideosService videosService;

    public VideosController(VideosService videosService) {
        this.videosService = videosService;
    }

    @GetMapping("/videos")
    public Object list(@Valid QueryVideosDto dto) {
        return videosService.findAll(dto);
    }

    @GetMapping("/videos/folders")
    public Object folders(@RequestParam(defaultValue = "") String parent) {
        return videosService.getFolderListing(parent);
    }

    @GetMapping("/videos/{id}")
    public Object one(@PathVariable int id) {
        return videosService.findOne(id);
    }

    @PatchMapping("/videos/{id}/tags")
    public Object patchTags(@PathVariable int id, @RequestBody UpdateVideoTagsDto body) {
        return videosService.updateTags(id, body.tagIds());
    }
}
